app/models/policy.rb

```diff
@@ -7,6 +7,14 @@ class Policy < ApplicationRecord
   validates :name, :version, :account_id, :schema, presence: true
   validate :belongs_to_a_tenant
   validate :validate_schema_specification
+  serialize :schema, Hash
+
+  def schema=(value)
+    json = value.is_a?(String) ? JSON.parse(value) : value
```

Comment:
it sounds simpler to me without the type check, like this:
```suggestion
    json = JSON.parse(value.to_s)
```
I don't think that we need to check if it is a String, if it is a wrong value, it will take it the rescue in line 15 anyway 😄

Reply:
Nice review! Many thanks